### ISSUE TITLE
Is training of batchnorm is always "False"

### ISSUE THREAD

**Issue by alldbi** [zhengyang-wang/Deeplab-v2--ResNet-101--Tensorflow#39]
Thank you for your great implementation. Can you please explain why "is_training" in all batch normalization layers are always "False"? This results in no updating of batch normalization parameters.

**Comment by zhengyang-wang** [comment]
Note that we are using transfer learning here. The encoder is a pre-trained model on ImageNet. The pre-training uses a much larger batch size (32, 64, 128, etc.) than the batch size here (10). The batch normalization relies on a large batch size. In addition, the number of images in ImageNet is much larger than that in PASCAL. As a result, we choose to fix the batch normalization parameters (gamma & beta).

**Comment by alldbi** [comment]
Thank you for your comment. This thing reduces the reusability of your code. So please add a comment to your main readme file and declare that for training your model from scratch, it is needed to change the batch normalization's training flag.

**Comment by zhengyang-wang** [comment]
Thanks for your suggestion. The reusability depends. For natural image segmentation on datasets like PASCAL, where transfer learning is applied, this is just a standard setting (Deeplab v2, R-CNN, etc.). Training from scratch (without pre-training) is not possible here.

If you have your own dataset, it may not be a good idea to use  such a deeper encoder unless you have enough training images. U-net may be a better choice.
